### PR TITLE
fix(DB/Creature): Scarlet Miners and Cars don't follow a path.

### PR DESCRIPTION
**Pull request by Rorschach91**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [X] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/20147

## SOURCE:

The changes have been validated through:
- [X] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [X] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [X] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [X] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Nyeriah — APPROVED** [review]

File: data/sql/updates/pending_db_world/Scarlet_miners_and_Cars_Fix.sql [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+-- Remove Wrong Guids
+
+DELETE FROM `creature` WHERE (`id1` = 28822) AND (`guid` IN (128885, 128886, 128887, 128888, 128889, 128890, 128891, 128897, 128900, 128902, 128903, 128904, 128905, 128906, 128907, 128908, 128909));
+DELETE FROM `creature` WHERE `id1` = 28821;
+DELETE FROM `creature_addon` WHERE (`guid` IN (128860, 128861, 128862, 128863, 128864, 128865, 128866, 128867, 128868, 128869, 128870, 128871, 128872, 128873, 128874, 128875, 128876, 128877, 128878, 128879, 128880, 128881, 128882, 128883, 128884));
+
+
+-- Add Waypoints
+
+DELETE FROM `waypoint_data` WHERE `id` IN (12889500, 12890100, 12889300, 12889400, 12889200, 12889800, 12889900, 12889600, 12889800);
```
Comment (sudlud) [inline data/sql/updates/pending_db_world/Scarlet_miners_and_Cars_Fix.sql:11]: double 12889800

File: data/sql/updates/pending_db_world/Scarlet_miners_and_Cars_Fix.sql [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+(28821, 0, 0, 0, 11, 0, 100, 0, 0, 0, 0, 0, 0, 0, 29, 1, 180, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 0, 'Mine Car - On Respawn - Start Follow Owner Or Summoner');
+
+
+-- Scarlet Miners
+
+-- Miners inside the Mine
+
+DELETE FROM `smart_scripts` WHERE (`source_type` = 0 AND `entryorguid` = -128895);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `event_param6`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(-128895, 0, 0, 1, 11, 0, 100, 0, 0, 0, 0, 0, 0, 0, 12, 28821, 8, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Scarlet Miner - On Respawn - Summon Creature \'Mine Car\''),
```
Comment (sudlud) [inline data/sql/updates/pending_db_world/Scarlet_miners_and_Cars_Fix.sql:183]: can cou please leave a comment like "has guid specific SAI" in the creature table for these guids?
this helps keeping an overview about entries with guid-specific SAI :)

Comment (Rorschach91) [reply data/sql/updates/pending_db_world/Scarlet_miners_and_Cars_Fix.sql:183]: Done.

**Comment by TheSCREWEDSoftware** [comment]
Just tested in-game, they are working as they should, now having the proper path in and out of the mine and from the mine into the beach

**Comment by TheSCREWEDSoftware** [comment]
@sudlud can you confirm the changes that you suggest is what you wanted (just the comment with the message) or with the actual GUIDs below?

**Comment by TheSCREWEDSoftware** [comment]
image

**Comment by Rorschach91** [comment]
> image

Maybe I missunderstanded what sudlud want, in any case all smartai under that comment are Guid SAI (only Mine Car is a general smartAI).

**Comment by sudlud** [comment]
The database table "creature" has a "comment" column. I'd like you to add a comment there right in the database. There should be some other occurrences like this already in the database.
Guid specific SAI is not happening too often so it's good to have a hint in the database right at the spawns that it's having a guid specific SAI for anyone wanting to touch / modify it in the future

**Comment by Rorschach91** [comment]
> The database table "creature" has a "comment" column. I'd like you to add a comment there right in the database. There should be some other occurrences like this already in the database. Guid specific SAI is not happening too often so it's good to have a hint in the database right at the spawns that it's having a guid specific SAI for anyone wanting to touch / modify it in the future

Ok, it is clear now. Done.

**Pull request merged**